syncer/syncer.go

```diff
@@ -204,6 +246,17 @@ func NewSyncer(
 	s.isBusy.Store(false)
 	s.lastLayerSynced.Store(s.mesh.LatestLayer().Uint32())
 	s.lastEpochSynced.Store(types.GetEffectiveGenesis().GetEpoch().Uint32() - 1)
+	if s.cfg.ReconcSync.Enable && s.asv2 == nil {
+		serverOpts := append(
+			s.cfg.ReconcSync.ServerConfig.ToOpts(),
+			server.WithHardTimeout(s.cfg.ReconcSync.HardTimeout))
+		s.dispatcher = sync2.NewDispatcher(s.logger, fetcher.(sync2.Fetcher), serverOpts)
```

Comment:
This cast shouldn't be necessary:

```suggestion
		s.dispatcher = sync2.NewDispatcher(s.logger, fetcher, serverOpts)
```

Also this here is the reason why the `sync2.Fetcher` interface needs to expose the `Host` method. I think instead it would be better to pass the host explicitly to `NewSyncer` and remove the method from the interface and `fetch.Fetch` implementation.

Reply:
Fixed by merging your branch